website/docs/docs/building-a-dbt-project/building-models/python-models.md

```diff
@@ -0,0 +1,560 @@
+---
+title: "Python models"
+---
+
+:::info Beta
+
+This feature is arriving in dbt Core v1.3, which is currently in beta. Note that only [specific data warehouses](#specific-data-warehouses) support dbt Python models, and that support for some dbt Cloud capabilities will be in progress during the beta period.
+
+We encourage you to:
+- [Read the original discussion](https://github.com/dbt-labs/dbt-core/discussions/5261)
+- [Weigh in on our developing best practices](https://github.com/dbt-labs/docs.getdbt.com/discussions/1811)
+- Join the **#beta-feedback-python-models** channel in the [dbt Community Slack](https://www.getdbt.com/community/join-the-community/)
+
+Below, you'll see sections entitled "❓ **Our questions**." We're working to develop our opinionated recommendations ahead of the final release this October—and you can help! Comment in the GitHub discussions, leave thoughts in Slack, talk about it with colleagues and friends.
+:::
+
+## What is a Python model in dbt?
+
+A dbt Python model is a function that reads in dbt sources or models, applies a series of transformations, and returns a transformed dataset. Just like SQL models!
+
+Unlike SQL models, which return a `select` statement, each Python model returns a single <Term id="dataframe">DataFrame</Term>. When you run a Python model, the result of that DataFrame will be saved as a table in your data warehouse.
+
+dbt Python models have access to almost all of the same configuration options as SQL models. You can test them, document them, add `tags` and `meta` properties to them, grant access on their results to other users, and so on. You can select them by their name, their file path, their configurations, whether they are upstream or downstream of another model, or whether they have been modified compared to a previous project state.
+
+### Defining a Python model
+
+Each Python model lives in a `.py` file in your `models/` folder. It defines a function named **`model()`**, which has two parameters:
+- **`dbt`**: A class compiled by dbt Core, unique to each model, that enables you to run your Python code in the context of your dbt project and DAG.
+- **`session`**: A class representing the connection to the backing engine, which allows you to interact with the data platform. The session is needed to read in tables as DataFrames, and to write DataFrames back to tables. In PySpark, by convention, the `SparkSession` is named `spark`, and available globally. For consistency across platforms, we always pass it into the `model` function as an explicit argument named `session`.
```

Comment:
The end user needs to access classmethods of `dbt`, and some properties of `session` as well. We think, better to document those and make it clearer where they're coming from, rather than pulling them out of the aether.

We could have gone for a "script"-style approach, where `dbt` and `session` appear to be available "globally":
```python
def some_fn:
    ...
    
some_df = dbt.ref("some_model")
final_df = some_df.apply(some_fn)

final_df
```
Where really we just take users' code, and turn it into:
```python
def model(dbt, session):
    def some_fn:
        ...
    
    some_df = dbt.ref("some_model")
    final_df = some_df.apply(some_fn)

    return final_df  # we add this return?
```
It's still an option we could consider, if people like that a lot better! (@ChenyuLInx)